filter/patterns_storage.go

```diff
@@ -88,7 +88,10 @@ func (storage *PatternStorage) matchPatterns(metric *ParsedMetric) []string {
 	seriesByTagPatternIndex := storage.SeriesByTagPatternIndex.Load().(*SeriesByTagPatternIndex)
 
 	matchedPatterns := make([]string, 0)
-	matchedPatterns = append(matchedPatterns, patternIndex.MatchPatterns(metric.Name)...)
-	matchedPatterns = append(matchedPatterns, seriesByTagPatternIndex.MatchPatterns(metric.Name, metric.Labels)...)
+	if len(metric.Labels) == 0 {
```

Comment:
NIT

It would be better to create function e.g. `isMetricLabled` to autodocument this piece of code